### DISCUSSION TITLE
Use of `readonly` for API types

### DISCUSSION THREAD

**Discussion by evantrimboli** [mikro-orm/mikro-orm#2069] [General]
There are a number of places in the API where it uses `readonly` for array types, but not others. In most cases as far as I can see it doesn't use `readonly` for objects.

For the places where it's not used, is it intentional and we should expect those arrays/objects to be modified, for example:

```ts
const ids : readonly number[] = [1, 2, 3];
const fetched = await this.repo.find({
  id: {
    $in: ids // Fails
  }
});
```

If not, can readonly modifiers be used more liberally?

**Comment by B4nan** [comment]
Not really intentional, queries are definitely not mutated, feel free to PR things you cross.
